File: Data Structures & Algorithms/is-subsequence/submission-0.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i = 0, j = 0;
        int nS = s.size();
        int nT = t.size();
        while (i < nS && j < nT){
            if(s[i] == t[j]){
                i++;
                j++;
            }
            else {
                j++;
            }
        }
        if (i == nS){
            return true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        Stack<TreeNode> st1 = new Stack<>(), st2 = new Stack<>();
        if (root == null) return res;
        st1.push(root);
        
        while (!st1.isEmpty()){
            TreeNode cur = st1.pop();
            st2.push(cur);
            if (cur.left != null) st1.push(cur.left);
            if (cur.right != null) st1.push(cur.right);
        }

        while (!st2.isEmpty()){
            res.add(st2.pop().val);
        }
        return res;
    }
}